ftml/src/parsing/rule/impls/block/blocks/checkbox.rs

```diff
@@ -0,0 +1,56 @@
+/*
+ * parsing/rule/impls/block/blocks/checkbox.rs
+ *
+ * ftml - Library to parse Wikidot text
+ * Copyright (C) 2019-2021 Wikijump Team
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+use super::prelude::*;
+
+pub const BLOCK_CHECKBOX: BlockRule = BlockRule {
+    name: "block-checkbox",
+    accepts_names: &["checkbox"],
+    accepts_special: false,
+    accepts_newlines: false,
+    parse_fn,
+};
+
+fn parse_fn<'r, 't>(
+    log: &slog::Logger,
+    parser: &mut Parser<'r, 't>,
+    name: &'t str,
+    special: bool,
+    in_head: bool,
+) -> ParseResult<'r, 't, Element<'t>> {
+    debug!(
+        log,
+        "Parsing checkbox block";
+        "in-head" => in_head,
+        "name" => name,
+    );
+
+    assert_eq!(special, false, "Checkboxes don't allow special variant");
```

Comment:
A special variant could start pre-checked (or some other syntax for that)

Reply:
I'm not sure what you mean. The checks here are asserts (meaning the program will crash) as safety, since the block rule should've already checked if special blocks are accepted.

Reply:
Oh that's a good idea.